ui/component/textareaWithSuggestions/view.jsx

```diff
@@ -0,0 +1,330 @@
+  doResolveUris: (Array<string>) => void,
+  hideSuggestions?: boolean,
+  inputRef: any,
+  maxLength?: number,
+  name: string,
+  noTopSuggestion?: boolean,
+  placeholder?: string,
+  showMature: boolean,
+  canonicalSubscriptions: Array<string>,
+  type?: string,
+  value: any,
+  onChange: (any) => any,
+};
+
+export default function TextareaWithSuggestions(props: Props) {
+  const {
+    canonicalCreatorUri,
+    className,
+    canonicalCommentors,
+    doResolveUris,
+    hideSuggestions,
+    inputRef,
+    maxLength,
+    name,
+    noTopSuggestion,
+    placeholder,
+    showMature,
+    canonicalSubscriptions,
+    type,
+    value: commentValue,
+    onChange,
+  } = props;
+
+  const inputProps = { className, placeholder };
+
+  const comboboxListRef: ElementRef<any> = React.useRef();
+
+  const [suggestionValue, setSuggestionValue] = React.useState(undefined);
+  const [debouncedTerm, setDebouncedTerm] = React.useState('');
+  const [topSuggestion, setTopSuggestion] = React.useState('');
+  const [canonicalSearchUris, setCanonicalSearchUris] = React.useState([]);
+
+  const suggestionTerm = suggestionValue && suggestionValue.term;
+  const isUriFromTermValid = suggestionTerm && !REGEX.INVALID_URI.test(suggestionTerm.substring(1));
+
+  const additionalOptions = { isBackgroundSearch: false, [SEARCH_OPTIONS.CLAIM_TYPE]: SEARCH_OPTIONS.INCLUDE_CHANNELS };
+  const { results, loading } = useLighthouse(debouncedTerm, showMature, SEARCH_SIZE, additionalOptions, 0);
+  const stringifiedResults = JSON.stringify(results);
+
+  const shouldFilter = (uri, previousLists) =>
+    uri !== canonicalCreatorUri && (!previousLists || !previousLists.includes(uri));
+
+  const filteredCommentors = canonicalCommentors.filter((uri) => shouldFilter(uri));
+  const filteredSubs = canonicalSubscriptions.filter((uri) => shouldFilter(uri, filteredCommentors));
+  const filteredTop = shouldFilter(topSuggestion, [...filteredCommentors, ...filteredSubs]) && topSuggestion;
+  const filteredSearch =
+    canonicalSearchUris &&
+    canonicalSearchUris.filter((uri) => shouldFilter(uri, [...filteredCommentors, ...filteredSubs, filteredTop || '']));
+
+  const creatorUriMatch = useSuggestionMatch(suggestionTerm || '', [canonicalCreatorUri]);
+  const subscriptionsMatch = useSuggestionMatch(suggestionTerm || '', filteredSubs);
+  const commentorsMatch = useSuggestionMatch(suggestionTerm || '', filteredCommentors);
+  const emotesMatch = useSuggestionMatch(
+    (suggestionValue && suggestionValue.isEmote && suggestionTerm) || '',
+    EMOTES.map(({ name }) => name)
+  );
+
+  const hasMinSearchLength = suggestionTerm && suggestionTerm.length >= LIGHTHOUSE_MIN_CHARACTERS;
+  const isTyping = suggestionValue && debouncedTerm !== suggestionValue.term;
+  const showPlaceholder = hasMinSearchLength && (isTyping || loading);
+
+  /** --------- **/
+  /** Functions **/
+  /** --------- **/
+
+  function handleChange(e: SyntheticInputEvent<*>) {
```

Comment:
I typed "hello world :angry @" 
On the "@", I was expecting channel suggestions to appear, but it stayed at emote suggestion.